packages/web-components/src/components/va-alert-expandable/va-alert-expandable.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  Listen,
+  State,
+  h,
+} from '@stencil/core';
+import classnames from 'classnames';
+
+@Component({
+  tag: 'va-alert-expandable',
+  styleUrl: 'va-alert-expandable.css',
+  shadow: true,
+})
+export class VaAlertExpandable {
+  @Element() el: HTMLElement;
+
+  @State() open: boolean;
+
+  /**
+   * Determines the icon and background color.
+   * One of `info`, `error`, `success`, `warning`, or `continue`
+   */
+  @Prop() status: string = 'info';
+
+  /**
+   * The text to trigger the expansion
+   */
+  @Prop() trigger: string = '';
+
+  /**
+   * If true, doesn't fire the CustomEvent which can be used for analytics tracking.
+   */
+  @Prop() disableAnalytics: boolean = false;
+
+  /**
+   * If false, the status icon is removed.
+   */
+  @Prop() showIcon: boolean = true;
+
+  /**
+   * If true, the alert will be visible.
+   */
+   @Prop() visible: boolean = true;
+
+  /**
+   * Fires when the component has successfully finished rendering for the first
+   * time.
+   */
+   @Event({
+    eventName: 'va-component-did-load',
+    composed: true,
+    bubbles: true,
+  })
+  vaComponentDidLoad: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link is clicked and disableAnalytics is not true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @Listen('resize', { target: 'window' })
+  handleResize() {
+    this.updateAlertBodyMaxHeight();
+  }
+
+  toggleOpen(): void {
+    // Conditionally track the event.
+    if (!this.disableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-alert-expandable',
+        action: !this.open ? 'expand' : 'collapse',
+        details: {
+          triggerText: this.trigger,
+        },
+      });
+    }
+
+    this.open = !this.open;
+  }
+
+  handleKeydown(event): void {
+    if (event.key === ' ' || event.key === 'Enter') {
+      event.preventDefault();
+      this.toggleOpen();
+    }
+  }
+
+  // Ensures that the CSS animation is consistent and uses the correct max-height for its content
+  updateAlertBodyMaxHeight() {
+    const bodyElm = this.el.shadowRoot.getElementById('alert-body');
+    const contentHeight = bodyElm.scrollHeight + 'px';
+    // the additional 2em is #alert-body margin-top and margin-bottom when open
+    bodyElm.style.setProperty(
+      '--calc-max-height',
+      'calc(' + contentHeight + ' + 2rem)',
+    );
+  }
+
+  componentDidLoad() {
+    this.updateAlertBodyMaxHeight();
+    this.vaComponentDidLoad.emit();
```

Comment:
Why is this happening?

Reply:
This emit is in `va-alert` as well, so I added it in here.